Userland/Applications/PixelPaint/Image.cpp

```diff
@@ -677,8 +677,8 @@ ErrorOr<void> Image::resize(Gfx::IntSize new_size, Gfx::Painter::ScalingMode sca
         if (layer->is_selected())
             selected_layer_index = i;
 
-        Gfx::IntPoint new_location(scale_x * new_layer->location().x(), scale_y * new_layer->location().y());
-        TRY(new_layer->resize(new_size, new_location, scaling_mode, Layer::NotifyClients::No));
+        auto new_rect = new_layer->relative_rect().to_type<float>().scaled(scale_x, scale_y).to_type<int>();
```

Comment:
`to_type<int>()` will always truncate the new size, right? Would it be more appropriate to round up? If we get a new size of e.g 1.999x1.999, I would expect the layer to be 2x2, not 1x1.

Reply:
Yes, rounding gives a better result in this case. I written an explanation of the new method I'm using to scale each layer in my comment below.